sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ReceiverManager.cs

```diff
@@ -203,6 +203,7 @@ protected async Task ProcessOneMessageWithinScopeAsync(ServiceBusReceivedMessage
                 ServiceBusFailureReason? failureReason = (ex as ServiceBusException)?.Reason;
                 if (!message.IsSettled &&
                     _receiverOptions.ReceiveMode == ServiceBusReceiveMode.PeekLock &&
+                    ProcessorOptions.AutoCompleteMessages &&
```

Comment:
I don't think this change is necessary since we are already checking that message.IsSettled is false. This means that if a user Deadletters the message, and throws an exception (which was the scenario in both linked issues), the message will not be abandoned. Please let me know if I am misunderstanding the intent here.

Reply:
I understand what you mean. However, without this change, then the behavior of `ReceiverManager` will become inconsistent with the behavior of `MessageReceivePump` and `SessionReceivePump` in this PR:
* In the pumps, if an exception is thrown without manually settling the message and AutoComplete is false, then the message lock will remain until it times out.
* In `ReceiverManager`, if an exception is thrown without manually settling the message and AutoComplete is false, then not including this change will have the message automatically abandoned.

I don't have a particular opinion on whether or not a message should be automatically abandoned if an exception is thrown without manually settling the message and AutoComplete is false, other than that I'd expect it to be consistent across these different handlers. So far as I can tell, there's not a means for a similar "IsSettled" check in the pumps (save perhaps for attempting to abandon it and only logging/raising the exception if AutoComplete is true).

Let me know your preference in light of this and I can update this PR to accomodate.

Reply:
We don't have a requirement that the behavior is the same between the two libraries here. Additionally, changing it now would be a behavior breaking change so I don't think we can do it. I think the same would be true for Microsoft.Azure.ServiceBus, but would have to defer to @shankarsama on that.

Reply:
I've reverted this change to `Azure.Messaging.ServiceBus.ReceiverManager`.

Would it be acceptable if I update the documentation for `AutoCompleteMessage` from:

```
If the message handler triggers an exception, the message will not be automatically completed.
```
to

```
If the message handler triggers an exception and did not settle the message,
then the message will be automatically abandoned, irrespective of <see cref="AutoCompleteMessages"/>.
```
?

It's maybe a bit verbose for a property description, but I couldn't find any other documentation relevant to this behavior.

Reply:
I think adding this additional information would be great. Rather than update existing line, I would just add the part about abandoning in a `<remarks>` tag. Thanks @zamzowd!